src/net_processing.cpp

```diff
@@ -1252,6 +1253,33 @@ bool PeerManagerImpl::BlockRequestAllowed(const CBlockIndex* pindex)
            (GetBlockProofEquivalentTime(*pindexBestHeader, *pindex, *pindexBestHeader, m_chainparams.GetConsensus()) < STALE_RELAY_AGE_LIMIT);
 }
 
+bool PeerManagerImpl::FetchBlock(const NodeId nodeid, const CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
```

Comment:
I don't think this `EXCLUSIVE_LOCKS_REQUIRED(cs_main)` is doing anything. This function is being called without `cs_main` (by `getblockfrompeer()`).

I think it's better to remove this annotation and lock `cs_main` within this function.